osu.Game.Rulesets.Osu/Difficulty/Skills/Speed.cs

```diff
@@ -32,18 +32,18 @@ public Speed(Mod[] mods)
         {
         }
 
-        private double strainDecay(double ms) => Math.Pow(strainDecayBase, ms / 1000);
+        protected override double CalculateInitialStrain(double time, DifficultyHitObject current) => (CurrentStrain * currentRhythm) * StrainDecay(time - current.Previous(0).StartTime);
 
-        protected override double CalculateInitialStrain(double time, DifficultyHitObject current) => (currentStrain * currentRhythm) * strainDecay(time - current.Previous(0).StartTime);
+        protected override double StrainValueOf(DifficultyHitObject current) => SpeedEvaluator.EvaluateDifficultyOf(current);
 
         protected override double StrainValueAt(DifficultyHitObject current)
         {
-            currentStrain *= strainDecay(((OsuDifficultyHitObject)current).StrainTime);
-            currentStrain += SpeedEvaluator.EvaluateDifficultyOf(current) * skillMultiplier;
+            CurrentStrain *= StrainDecay(((OsuDifficultyHitObject)current).StrainTime);
+            CurrentStrain += StrainValueOf(current) * SkillMultiplier;
 
             currentRhythm = RhythmEvaluator.EvaluateDifficultyOf(current);
 
-            double totalStrain = currentStrain * currentRhythm;
+            double totalStrain = CurrentStrain * currentRhythm;
```

Comment:
worth noting this entire thing becomes a little more confusing to read with these changes, but it's not really at fault of this PR - rhythm really should be entirely separate or atleast additive to speed so that we don't need to do this stupid solution. not a blocker, just speaking my mind